jack-core/src/com/rapleaf/jack/transaction/DbPoolManager.java

```diff
@@ -110,6 +120,33 @@ public void close() {
     connectionPool.close();
   }
 
+  @Override
+  public DbMetrics getMetrics() {
+    if (metricsTrackingEnabled) {
+      metrics.update(false, connectionPool);
+      return metrics;
+    } else {
+      LOG.info("logging disabled, db metrics cannot be accessed");
+      return null;
+    }
```

Comment:
Since the `metrics` is always constructed in the transactor, to avoid null pointer exception, you can still return `metrics` even when tracking is disabled. Just that all the method call from the return `metrics` will give zero, which is fine.